### PR TITLE
docs: add docs

### PR DESCRIPTION
**Pull request by noahwaldner**

### CODE REVIEW

File: .github/workflows/fondue-continuous-integration.yml [hunk 1]
```diff
@@ -4,6 +4,8 @@ on:
     pull_request:
         paths:
             - packages/fondue/**
+            - packages/components/**
+            - packages/docs/**
```
Comment (SamuelAlev) [inline .github/workflows/fondue-continuous-integration.yml:8]:
```suggestion
            - docs/**
```

File: .changeset/pretty-trains-camp.md
Comment (SamuelAlev) [inline .changeset/pretty-trains-camp.md:1]: not necessary as we don't want to bump version anywhere

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# React + TypeScript + Vite
```
Comment (SamuelAlev) [inline docs/README.md:1]: Please replace with another readme or delete the file

File: docs/package.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{
+    "name": "docs",
+    "private": true,
+    "version": "0.0.0",
+    "type": "module",
+    "files": [
+        "dist",
+        "tailwind.config.js"
+    ],
```
Comment (SamuelAlev) [inline docs/package.json:9]:
```suggestion
```

File: docs/package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    "scripts": {
+        "dev": "vite",
+        "build": "tsc && vite build",
+        "lint": "eslint .",
+        "lint:fix": "eslint --fix .",
+        "lint:prettier": "prettier --check .",
```
Comment (SamuelAlev) [inline docs/package.json:15]:
```suggestion
```
Included in the eslint checks

File: docs/package.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
+{
+    "name": "docs",
```
Comment (SamuelAlev) [inline docs/package.json:2]:
```suggestion
    "name": "@fondue/docs",
```

File: docs/package.json [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        "@tailwindcss/forms": "^0.5.3",
+        "components": "workspace:*",
+        "mdx-bundler": "^10.0.1",
+        "react": "^18.2.0",
+        "react-dom": "^18.2.0",
+        "vike": "^0.4.160",
+        "vike-react": "^0.4.4"
+    },
+    "devDependencies": {
+        "@frontify/fondue-tokens": "^3.4.0",
```
Comment (SamuelAlev) [inline docs/package.json:29]: Should go to `dependencies`

File: docs/tailwind.config.js
Comment (SamuelAlev) [inline docs/tailwind.config.js:1]: Rename the file to `tailwind.config.ts` and add the typing https://tailwindcss.com/docs/configuration#using-esm-or-type-script

File: docs/src/pages/index/+Page.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { allComponents } from 'components/content';
+export { Page };
+
+function Page() {
```
Comment (SamuelAlev) [inline docs/src/pages/index/+Page.tsx:6]:
```suggestion

export const Page = () => {
```
Code style of the codebase

File: docs/src/pages/component/+route.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+export { route };
```
Comment (SamuelAlev) [inline docs/src/pages/component/+route.ts:3]:
```suggestion
```

File: packages/components/.gitignore [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+!.vscode/extensions.json
+.idea
+.DS_Store
+*.suo
+*.ntvs*
+*.njsproj
+*.sln
+*.sw?
+
+.contentlayer
```
Comment (SamuelAlev) [inline packages/components/.gitignore:26]: Please add EOF

File: packages/components/package.json [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    "files": [
+        "dist",
+        "tailwind.config.js"
+    ],
+    "main": "./dist/component-library.cjs",
+    "module": "./dist/component-library.js",
+    "exports": {
+        "./content": {
+            "import": "./.contentlayer/generated/index.mjs",
+            "types": "./.contentlayer/generated/index.d.ts"
```
Comment (SamuelAlev) [inline packages/components/package.json:15]:
```suggestion
            "types": "./.contentlayer/generated/index.d.ts"
            "import": "./.contentlayer/generated/index.mjs",
```

Same for "."

File: packages/components/package.json [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+    },
+    "scripts": {
+        "dev": "tsc && concurrently 'vite build --watch' 'contentlayer dev'",
+        "build": "tsc && pnpm build:lib && contentlayer build ",
+        "build:lib": "vite build",
+        "preview": "vite preview"
+    },
+    "sideEffects": [
+        "**/*.css"
+    ],
```
Comment (SamuelAlev) [inline packages/components/package.json:32]: Why do you need to declare css file as side-effect?

File: packages/components/package.json [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+    "devDependencies": {
```
Comment (SamuelAlev) [inline packages/components/package.json:33]: `dependencies` before `devDependencies` (if the linter is correctly configured, it should complains already)

File: packages/components/package.json [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+        "@types/react": "^18.2.51",
+        "typescript": "^5.2.2",
```
Comment (SamuelAlev) [inline packages/components/package.json:35]: missing eslint, prettier, frontify eslint config, prettierrc file

File: packages/components/package.json [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+        "vite": "^5.0.8",
+        "vite-plugin-lib-inject-css": "^1.3.0"
```
Comment (SamuelAlev) [inline packages/components/package.json:37]: Do we really want to have the css file injected via JS? I would prefer a different export as it is in the current fondue version

File: packages/components/src/index.ts [hunk 15]
```diff
@@ -0,0 +1,4 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import './styles.css';
+export * from './components';
```
Comment (SamuelAlev) [inline packages/components/src/index.ts:4]:
```suggestion
export { ... } from './components';
```
use named export please

File: packages/components/tailwind.config.js [hunk 16]
```diff
@@ -0,0 +1,132 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+/** @type {import('tailwindcss').Config} */
+
+module.exports = {
```
Comment (SamuelAlev) [inline packages/components/tailwind.config.js:5]:
```suggestion
/** @type {import('tailwindcss').Config} */
module.exports = {
```

File: packages/components/vite.config.ts [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+import { defineConfig } from 'vite';
+import dts from 'vite-plugin-dts';
+
+export default defineConfig({
+    plugins: [dts({ insertTypesEntry: true, rollupTypes: true })],
+    build: {
+        cssCodeSplit: true,
+        outDir: './dist/lib',
+        lib: {
+            // Could also be a dictionary or array of multiple entry points
```
Comment (SamuelAlev) [inline packages/components/vite.config.ts:13]:
```suggestion
```

File: packages/components/vite.config.ts [hunk 18]
```diff
@@ -0,0 +1,33 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { resolve } from 'path';
+import { defineConfig } from 'vite';
+import dts from 'vite-plugin-dts';
+
+export default defineConfig({
+    plugins: [dts({ insertTypesEntry: true, rollupTypes: true })],
```
Comment (SamuelAlev) [inline packages/components/vite.config.ts:8]:
```suggestion
    plugins: [react(), dts({ insertTypesEntry: true, rollupTypes: true })],
```
missing the vite plugin react?

File: packages/components/vite.config.ts [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+    build: {
+        cssCodeSplit: true,
+        outDir: './dist/lib',
+        lib: {
+            // Could also be a dictionary or array of multiple entry points
+            entry: resolve(__dirname, 'src/index.ts'),
+            formats: ['es'],
+            // the proper extensions will be added
```
Comment (SamuelAlev) [inline packages/components/vite.config.ts:16]:
```suggestion
```

File: packages/components/vite.config.ts [hunk 20]
```diff
@@ -0,0 +1,33 @@
...
+        rollupOptions: {
+            // make sure to externalize deps that shouldn't be bundled
+            // into your library
+            external: ['react'],
+            output: {
+                // Provide global variables to use in the UMD build
+                // for externalized deps
+                globals: {
+                    react: 'React',
+                },
```
Comment (SamuelAlev) [inline packages/components/vite.config.ts:28]:
```suggestion
            external: ['react'],
            output: {
                globals: {
                    react: 'React',
                },
```
Please use something like this for `external` and `globals` and remove the comments

File: docs/package.json (hunk 3 above)
Comment (SamuelAlev) [reply docs/package.json:9]: Not solved, we don't need to expose

File: docs/package.json (hunk 4 above)
Comment (SamuelAlev) [reply docs/package.json:15]: Not solved, we don't need it

File: docs/src/renderer/types.ts [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+
+import { Component } from 'components/content';
+
+// https://vike.dev/pageContext#typescript
+declare global {
+    namespace Vike {
+        interface PageContext {
+            Page: () => React.ReactElement;
+            data?: {
+                /** Value for <title> defined dynmically by by /pages/some-page/+data.js */
```
Comment (SamuelAlev) [inline docs/src/renderer/types.ts:12]:
```suggestion
                /** Value for <title> defined dynamically by /pages/some-page/+data.ts */
```

File: docs/src/renderer/types.ts [hunk 22]
```diff
@@ -0,0 +1,31 @@
...
+                title?: string;
+                /** Value for <meta name="description"> defined dynmically */
+                description?: string;
+                component: Component;
+            };
+            config: {
+                /** Value for <title> defined statically by /pages/some-page/+title.js (or by `export default { title }` in /pages/some-page/+config.js) */
```
Comment (SamuelAlev) [inline docs/src/renderer/types.ts:19]:
```suggestion
                /** Value for <title> defined statically by /pages/some-page/+title.ts (or by `export default { title }` in /pages/some-page/+config.ts) */
```

File: docs/src/renderer/types.ts [hunk 23]
```diff
@@ -0,0 +1,31 @@
...
+                /** Value for <meta name="description"> defined statically */
+                description?: string;
+            };
+            /** https://vike.dev/render */
+            abortReason?: string;
+        }
+    }
+}
+
+// Tell TypeScript this file isn't an ambient module
```
Comment (SamuelAlev) [inline docs/src/renderer/types.ts:30]:
```suggestion
```

File: docs/tailwind.config.ts [hunk 24]
```diff
@@ -0,0 +1,133 @@
...
+
+export default {
+    presets: [require('@frontify/fondue-tokens/tailwind')],
+    content: ['./src/**/*.{js,ts,tsx}'],
+    plugins: [
+        tailwindForms({
+            strategy: 'class',
+        }),
+    ],
+    theme: {
```
Comment (SamuelAlev) [inline docs/tailwind.config.ts:14]: why do we need the theme, isn't `@frontify/fondue-tokens/tailwind` enough?

File: packages/components/.eslintrc [hunk 25]
```diff
@@ -0,0 +1,26 @@
+{
+    "root": true,
+    "extends": ["@frontify/eslint-config-react", "plugin:jsx-a11y/recommended", "plugin:storybook/recommended"],
```
Comment (SamuelAlev) [inline packages/components/.eslintrc:3]:
```suggestion
    "extends": ["@frontify/eslint-config-react", "plugin:jsx-a11y/recommended"],
```

File: packages/components/package.json
Comment (SamuelAlev) [inline packages/components/package.json:1]: missing frontify eslint config react, eslint plugin a11y and eslint plugin notice

File: docs/.eslintrc [hunk 26]
```diff
@@ -0,0 +1,26 @@
+{
+    "root": true,
+    "extends": ["@frontify/eslint-config-react", "plugin:jsx-a11y/recommended", "plugin:storybook/recommended"],
```
Comment (SamuelAlev) [inline docs/.eslintrc:3]:
```suggestion
    "extends": ["@frontify/eslint-config-react", "plugin:jsx-a11y/recommended"],
```

File: packages/components/src/components/Tag/Tag.tsx [hunk 27]
```diff
@@ -0,0 +1,8 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+type TagProps = {
+    children: React.ReactNode;
+};
+export const Tag = ({ children }: TagProps) => {
```
Comment (SamuelAlev) [inline packages/components/src/components/Tag/Tag.tsx:6]:
```suggestion
};

export const Tag = ({ children }: TagProps) => {
```
Let's bring some air 🌬️

File: packages/components/src/components/Tag/Tag.tsx [hunk 28]
```diff
@@ -0,0 +1,8 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+type TagProps = {
+    children: React.ReactNode;
```
Comment (SamuelAlev) [inline packages/components/src/components/Tag/Tag.tsx:4]: add `import { type ReactNode } from 'react';`
```suggestion
    children: ReactNode;
```

File: packages/components/tsconfig.json [hunk 29]
```diff
@@ -0,0 +1,29 @@
...
+
+        /* Bundler mode */
+        "moduleResolution": "bundler",
+        "allowImportingTsExtensions": true,
+        "resolveJsonModule": true,
+        "isolatedModules": true,
+        "noEmit": true,
+
+        /* Linting */
+        "strict": true,
```
Comment (SamuelAlev) [inline packages/components/tsconfig.json:17]:
```suggestion
        "strict": true,
        "noUncheckedIndexedAccess": true,
```

File: packages/components/package.json [hunk 30]
```diff
@@ -0,0 +1,54 @@
+{
+    "name": "@frontify/fondue-components",
+    "private": true,
+    "version": "0.0.0",
+    "type": "module",
+    "files": [
+        "dist",
+        "tailwind.config.js"
```
Comment (SamuelAlev) [inline packages/components/package.json:8]:
```suggestion
        "tailwind.config.ts"
```

File: packages/components/vite.config.ts [hunk 31]
```diff
@@ -0,0 +1,28 @@
...
+import { resolve } from 'path';
+import { defineConfig } from 'vite';
+import dts from 'vite-plugin-dts';
+import react from '@vitejs/plugin-react';
+
+export default defineConfig({
+    plugins: [react(), dts({ insertTypesEntry: true, rollupTypes: true })],
+    build: {
+        cssCodeSplit: true,
+        outDir: './dist/lib',
```
Comment (SamuelAlev) [inline packages/components/vite.config.ts:12]:
```suggestion
        outDir: './dist',
```

File: packages/components/package.json [hunk 32]
```diff
@@ -0,0 +1,54 @@
...
+    "exports": {
+        "./content": {
+            "types": "./.contentlayer/generated/index.d.ts",
+            "import": "./.contentlayer/generated/index.mjs"
+        },
+        ".": {
+            "types": "./dist/lib/component-lib.d.ts",
+            "import": "./dist/lib/component-lib.js"
+        },
+        "./styles": "./dist/lib/index.css",
```
Comment (SamuelAlev) [inline packages/components/package.json:21]:
```suggestion
            "types": "./dist/component-lib.d.ts",
            "import": "./dist/component-lib.js"
        },
        "./styles": "./dist/index.css",
```

File: packages/components/package.json [hunk 33]
```diff
@@ -0,0 +1,54 @@
...
+        "./tailwind": "./tailwind.config.js"
+    },
+    "scripts": {
+        "dev": "tsc && concurrently 'vite build --watch' 'contentlayer dev'",
```
Comment (SamuelAlev) [inline packages/components/package.json:25]:
```suggestion
        "dev": "concurrently 'vite build --watch' 'contentlayer dev'",
```

File: packages/components/package.json [hunk 34]
```diff
@@ -0,0 +1,54 @@
...
+        "build": "tsc && pnpm build:lib && contentlayer build ",
```
Comment (SamuelAlev) [inline packages/components/package.json:26]:
```suggestion
        "build": "tsc && pnpm build:lib && contentlayer build",
```

File: packages/components/vite.config.ts
Comment (SamuelAlev) [inline packages/components/vite.config.ts:1]: I would actually not bundle the packages and rather expose directly the TS files (then only the `docs` and `fondue` would have a build step)

Comment (SamuelAlev) [reply packages/components/vite.config.ts:1]: you can in the package.json use `main: './src/index.ts',` then we don't need a double dev "watch" and neither a build step when you update a component, happy to show you what I mean

**Review by SamuelAlev — CHANGES_REQUESTED** [review]

File: docs/package.json (hunk 3 above)
Comment (noahwaldner) [reply docs/package.json:9]: what exactly is the change here? github shows the exact sams snippe twice

Comment (SamuelAlev) [reply docs/package.json]: It is deleting the `files` array as we don't need to expose those files (it is not a npm package to publish)

File: docs/package.json [hunk 35]
```diff
@@ -0,0 +1,41 @@
...
+        "@tailwindcss/forms": "^0.5.3",
+        "mdx-bundler": "^10.0.1",
+        "react": "^18.2.0",
+        "react-dom": "^18.2.0",
+        "vike": "^0.4.160",
+        "vike-react": "^0.4.4"
+    },
+    "devDependencies": {
+        "@types/react": "^18.2.43",
```
Comment (SamuelAlev) [inline docs/package.json:25]: Missing the eslint plugins...
```
 "extends": ["@frontify/eslint-config-react", "plugin:jsx-a11y/recommended"],
    "plugins": ["jsx-a11y", "notice"],
```

File: docs/public/nook-animated.png
Comment (SamuelAlev) [inline docs/public/nook-animated.png:1]: non animated version please

File: docs/src/pages/component/+data.ts [hunk 36]
```diff
@@ -0,0 +1,23 @@
...
+
+export const data = async (pageContext: PageContextServer) => {
+    const { routeParams } = pageContext;
+
+    const component = allComponents.find((component) => {
+        return component.url === routeParams.pageUrl;
+    });
+
+    if (!component) {
+        throw render(404, "This Component Doesn't exist");
```
Comment (SamuelAlev) [inline docs/src/pages/component/+data.ts:17]:
```suggestion
        throw render(404, "This component doesn't exist");
```

File: docs/src/pages/component/+route.ts [hunk 37]
```diff
@@ -0,0 +1,18 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { render } from 'vike/abort';
+import type { RouteSync } from 'vike/types';
+
+export const route: RouteSync = (pageContext): ReturnType<RouteSync> => {
+    const pageUrl = pageContext.urlPathname.toLocaleLowerCase().split('/component/').pop();
+
+    if (!pageUrl) {
+        throw render(404, "This Component doesn't exist");
```
Comment (SamuelAlev) [inline docs/src/pages/component/+route.ts:10]:
```suggestion
        throw render(404, "This component doesn't exist");
```

File: packages/components/src/components/Tag/index.ts
Comment (SamuelAlev) [inline packages/components/src/components/Tag/index.ts:1]: Not needed as we export in `packages/components/src/index.ts` (we don't use anymore barrel exports where possible)

File: .github/workflows/fondue-continuous-integration.yml [hunk 38]
```diff
@@ -4,6 +4,8 @@ on:
     pull_request:
         paths:
             - packages/fondue/**
+            - packages/components/**
+            - docs/**
```
Comment (SamuelAlev) [inline .github/workflows/fondue-continuous-integration.yml:8]: I would split the job for the `docs` to be it own one, wdyt?

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**